nixos/modules/services/web-apps/healthchecks.nix

```diff
@@ -0,0 +1,216 @@
+        If you close new user registration, you can still selectively invite users to your team account.
+      '';
+    };
+
+    extraConfig = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+        Additional lines of configuration appended to the <literal>local_settings.py</literal>.
+        See the <link xlink:href="https://healthchecks.io/docs/self_hosted_configuration/">documentation</link> for more possible options.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ healthchecksManageScript ];
+
+    systemd.targets.healthchecks = {
+      description = "Target for all Healthchecks services";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network-online.target"];
+    };
+
+    systemd.services = let
+      defaultServiceConfig = {
+        WorkingDirectory = "${cfg.dataDir}";
+        User = "healthchecks";
+        Group = "healthchecks";
+        StateDirectory = "healthchecks";
+        StateDirectoryMode = "0750";
+        Restart = "on-failure";
+      };
+    in {
+      healthchecks-migration = {
+        description = "Healthchecks migrations";
+        wantedBy = [ "healthchecks.target" ];
+
+        environment = {
+          PYTHONPATH = pkg.pythonPath;
+        };
+
+        serviceConfig = defaultServiceConfig // {
+          Type = "oneshot";
+          ExecStart = ''
+            ${pkg}/bin/healthchecks migrate
+          '';
+        };
+      };
+
+      healthchecks = {
+        description = "Healthchecks WSGI Service";
+        wantedBy = [ "healthchecks.target" ];
+        after = [ "healthchecks-migration.service" ];
+
+        preStart = ''
+          ${pkg}/bin/healthchecks collectstatic --no-input
+          ${pkg}/bin/healthchecks remove_stale_contenttypes --no-input
+          ${pkg}/bin/healthchecks compress
+        '';
+
+        environment = {
+          PYTHONPATH = pkg.pythonPath;
+        };
+
+        serviceConfig = defaultServiceConfig // {
+          ExecStart = ''
+            ${pkgs.python3Packages.gunicorn}/bin/gunicorn hc.wsgi \
+              --bind ${cfg.listenAddress}:${toString cfg.port} \
+              --pythonpath ${pkg}/opt/healthchecks
+          '';
+        };
+      };
+
+      healthchecks-sendalerts = {
+        description = "Healthchecks Alert Service";
+        wantedBy = [ "healthchecks.target" ];
+        after = [ "healthchecks.service" ];
+
+        environment = {
+          PYTHONPATH = pkg.pythonPath;
+        };
+
+        serviceConfig = defaultServiceConfig // {
+          ExecStart = ''
+           ${pkg}/bin/healthchecks sendalerts
+         '';
+        };
+      };
+
+      healthchecks-sendreports = {
+        description = "Healthchecks Reporting Service";
+        wantedBy = [ "healthchecks.target" ];
+        after = [ "healthchecks.service" ];
+
+        environment = {
+          PYTHONPATH = pkg.pythonPath;
+        };
```

Comment:
This shouldn't be required. Are we missing the wrapping for the binary maybe?

Reply:
You are right, it wasn't required for most of the services. But it is required for the healthchecks service as that one isn't simply run from the wrapper. We start gunicorn, which loads our WSGI service as a python module and therefore needs to find it in PYTHONPATH. using `pkg.pythonPath` for gunicorns `--pythonpath` flag should work in theory but does not in practice, presumably because the path gets to long. (systemd complained about a bad unit file)

Reply:
🤔 I think we could avoid this but it would require a patch and creating another python script. Since that is more maintenance work we can set the env for that one service.